getting_started/guides/godot_essentials_guide.rst

```diff
@@ -0,0 +1,602 @@
+
+You can create a new instance of a Resource...
+
+- in code by calling ``.new()`` on a Resource-deriving script.
+- in code by calling
+  :ref:`duplicate() <class_Resource_method_duplicate>``
+  on an existing Resource instance.
+- in the editor by clicking the "New" icon in the Inspector.
+- in the editor by duplicating a Resource file in the FileSystem dock.
+
+You can also create your own
+:ref:`custom Resource types <doc_resources_custom_resources>`.
+They are effective tools for designing and organizing your project's data
+structures. The linked documentation describes many more benefits.
+
+Godot Engine doesn't, by default, keep track of scripts' names. It recognizes
+them like any other Resource: by their file path. But, Godot 3.1 added the
+ability to define
+:ref:`script classes <doc_scripting_continued_class_name>`, i.e. global names
+for scripts. Only a few languages have script class support for now, but
+Godot will add more in the future.
+
+.. note::
+
+  C# has its own internal ability to define names for scripts and reference
+  them within other C# scripts. This is not the result of script classes.
+  Script classes would enable other languages and the editor itself to also
+  see C# class names.
+
+For more information on reference-counting, see
+`this article <https://mortoray.com/2012/01/08/what-is-reference-counting/>`__.
+
+For more information on how reference-counting
+compares to garbage collection, see
+`this article <https://medium.com/computed-comparisons/garbage-collection-vs-automatic-reference-counting-a420bd4c7c81>`__.
+
+The editor, tools, and plugins
+------------------------------
+
+One of the most empowering features Godot has is the fact that the
+editor itself is also built using Godot Engine. This is a matter of
+function more than marketability.
+
+Because the Godot Editor has a SceneTree, it is possible to inject
+your own Nodes into it. This makes the creation of custom tools
+straightforward with almost no learning curve.
+
+1. You can declare any script as a "tool" script. This enables
+   that script's logic to execute in the editor.
+
+   To specify editor-only or game-only logic, you can check the 
+   script's current execution mode with
+   :ref:`Engine.is_editor_hint <class_Engine_property_is_editor_hint>`.
+   
+   Each scripting language will have its own syntax for declaring tools. Please
+   consult the scripting documentation for your language of choice for more
+   information.
+
+2. If you need to change the editor itself, Godot provides an
+   easy-to-use
+   :ref:`plugin API <class_EditorPlugin>`. For one-off edits, you can also
+   write :ref:`standalone editor scripts <class_EditorScript>`.
+   For more information about how to create plugins,
+   :ref:`click here <doc_making_plugins>`.
+
+3. There's no need to learn a domain-specific language for plugins.
+   Instead, Godot enables you to re-use your game UI knowledge.
+
+4. While writing a plugin, you can also add
```

Comment:
Maybe also add a 5th point that elaborates briefly and links to other *Plugin subtypes such as importers and the like. Otherwise I only touch on the InspectorPlugins and nothing else.